dialogue-core/src/main/java/com/palantir/dialogue/core/CautiousIncreaseAggressiveDecreaseConcurrencyLimiter.java

```diff
@@ -76,7 +82,59 @@
         return Optional.empty();
     }
 
-    final class Permit implements FutureCallback<Response> {
+    enum Behavior {
+        HOST_LEVEL() {
+            @Override
+            void onSuccess(Response result, PermitControl control) {
+                if ((Responses.isQosStatus(result) && !Responses.isTooManyRequests(result))) {
+                    control.dropped();
+                } else if (Responses.isServerError(result) || Responses.isTooManyRequests(result)) {
```

Comment:
What do you think about promoting non-500 server errors to the host level, and only using 500 status server errors for per-endpoint limits?

Reply:
502 and 504 are relatively common in some environments, and they apply to the entire channel.

Reply:
I'd be OK with carving out 502 and 504 as we do commonly see them from nginx / proxies, but personally I'd actually keep 500 as an endpoint-level concern tbh, because I think it's very possible that someone writes a bug in their FooResource#myEndpoint method that wouldn't affect any other endpoints

Reply:
Yep, I think that's exactly what I'm suggesting, we apply the `500` status to individual endpoints, but other 5xx status codes (501 through 599 inclusive) to the host.